### PR TITLE
Generators generated

### PR DESCRIPTION
**Pull request by LtRipley36706**
* Overhauled Generator system to bring it more in line with how it operated on retail servers.
  * You'll need to load some test data included with this update in the Updates/World folder for this code to work.
  * Locations of the test generators are below. The creature generators do not support regeneration, while the item generators will regenerate.
  * /teleloc 0x7f0301ad [12.319899559021 -28.482000350952 0.0049999998882413] 0.33894598484039 0 0 0.94080585241318
  * /telepoi Holtburg

This is by no means the complete system but I think its a good step towards one than what is currently in ACE. Lots more to do no doubt.

### CODE REVIEW

File: Source/ACE/Entity/GenericObject.cs [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
+
+namespace ACE.Entity
 {
     public class GenericObject : WorldObject
     {
         public GenericObject(AceObject aceObject)
             : base(aceObject)
         {
+            if (AnimationFrame == null)
+                AnimationFrame = 101;
```
Comment (ddevec) [inline Source/ACE/Entity/GenericObject.cs]: What does this mean?  Can you put in a constant, or enum value instead, to make this clear?

File: Source/ACE/Entity/Landblock.cs [hunk 2]
```diff
@@ -843,6 +843,17 @@ private void ItemTransferContainerInternal(ObjectGuid woGuid, Container containe
             wo.Location = null;
             wo.InitializeAceObjectForSave();
             container.AddToInventory(wo, placement);
+
+            // Was Item controlled by a generator?
+
+            if (wo.GeneratorId > 0)
```
Comment (ddevec) [inline Source/ACE/Entity/Landblock.cs]: Should this be a function of the landblock, or a function of the item?

In other words, is it the landblock's responsibility to know that generators exist, and that objects may have generators, or is it the object's responsibility to notify the generator on pickup.  (And the landblock's responsibility to notify the item it was picked up).

IMO landblocks should have objects and be oblivious to what they are, or if they have generators.

**Comment by LtRipley36706** [comment]
Refined the generator system to make it easier to add content/generators and use existing database tables (with a minor alteration). This linking system also uses the guid generation system to avoid collisions.

File: Source/ACE/Entity/GenericObject.cs (hunk 1 above)
Comment (LtRipley36706) [reply Source/ACE/Entity/GenericObject.cs]: Need to apply some researched data to this, will fix shortly.

File: Source/ACE/Entity/Landblock.cs (hunk 2 above)
Comment (LtRipley36706) [reply Source/ACE/Entity/Landblock.cs]: More thinking to be done here, a TODO perhaps?

**Comment by LtRipley36706** [comment]
Addressed AnimationFrame aka Placement

**Pull request merged**